File: 01/haskell/01_2.hs
import System.Environment (getArgs)
import Helper (splitOn)
import Data.List (sort)

main :: IO ()
main = do
    args <- getArgs
    input <- readFile (head args)
    let elves = map (map read . lines) $ splitOn "\n\n" input :: [[Int]]
    print $ sum $ take 3 $ reverse $ sort $ map sum elves

File: 01/rust/01_2.rs
use std::env;
use std::fs;

fn main() {
	let mut elves = fs::read_to_string(env::args().nth(1).unwrap())
		.unwrap()
		.as_str()
		.split("\n\n")
		.map(|elf| elf
			.lines()
			.map(|c| c.parse::<i32>().unwrap())
			.sum::<i32>()
		)
		.collect::<Vec<_>>();

	elves.sort();

	println!("{}",
		elves
			.into_iter()
			.rev()
			.take(3)
			.sum::<i32>()
	);
}